go-controller/pkg/ovn/kubevirt.go

```diff
@@ -46,18 +53,261 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
+				matches := item.IPPrefix == clusterSubnetRoute.IPPrefix && item.Nexthop == clusterSubnetRoute.Nexthop && item.Policy != nil && *item.Policy == *clusterSubnetRoute.Policy
+				return matches
+			}); err != nil {
+				return fmt.Errorf("failed adding static route: %v", err)
+			}
+		}
+	}
+
+	for _, podIP := range podAnnotation.IPs {
+		podAddress := podIP.IP.String()
+		// In case of one global zone a policy is needed to route traffic
+		// to the gateway
+		if !config.OVNKubernetesFeature.EnableInterconnect {
+			ipFamily := utilnet.IPFamilyOfCIDR(podIP)
+			nodeGwAddress, err := util.MatchFirstIPNetFamily(ipFamily == utilnet.IPv6, lrpAddresses)
+			if err != nil {
+				return err
+			}
+			match := fmt.Sprintf("ip%s.src == %s", ipFamily, podAddress)
+			egressPolicy := nbdb.LogicalRouterPolicy{
+				Match:    match,
+				Action:   nbdb.LogicalRouterPolicyActionReroute,
+				Nexthops: []string{nodeGwAddress.IP.String()},
+				Priority: 1,
+				ExternalIDs: map[string]string{
+					kubevirt.NamespaceExternalIDKey:    pod.Namespace,
+					kubevirt.OvnZoneExternalIDKey:      kubevirt.OvnLocalZone,
+					kubevirtv1.VirtualMachineNameLabel: pod.Labels[kubevirtv1.VirtualMachineNameLabel],
+				},
+			}
+			if err := libovsdbops.CreateOrUpdateLogicalRouterPolicyWithPredicate(oc.nbClient, types.OVNClusterRouter, &egressPolicy, func(item *nbdb.LogicalRouterPolicy) bool {
+				return item.Priority == egressPolicy.Priority && item.Match == egressPolicy.Match && item.Action == egressPolicy.Action
+			}); err != nil {
+				return fmt.Errorf("failed adding point to point policy for pod %s/%s : %v", pod.Namespace, pod.Name, err)
+			}
+		}
+
+		// Add a route for reroute ingress traffic to to the VM port since
+		// the subnet is alien to ovn_cluster_router
+		outputPort := types.RouterToSwitchPrefix + pod.Spec.NodeName
+		ingressRoute := nbdb.LogicalRouterStaticRoute{
+			IPPrefix:   podAddress,
+			Nexthop:    podAddress,
+			Policy:     &nbdb.LogicalRouterStaticRoutePolicyDstIP,
+			OutputPort: &outputPort,
+			ExternalIDs: map[string]string{
+				kubevirt.NamespaceExternalIDKey:    pod.Namespace,
+				kubevirt.OvnZoneExternalIDKey:      kubevirt.OvnLocalZone,
+				kubevirtv1.VirtualMachineNameLabel: pod.Labels[kubevirtv1.VirtualMachineNameLabel],
+			},
+		}
+		if err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(oc.nbClient, types.OVNClusterRouter, &ingressRoute, func(item *nbdb.LogicalRouterStaticRoute) bool {
+			matches := item.IPPrefix == ingressRoute.IPPrefix && item.Nexthop == ingressRoute.Nexthop && item.Policy != nil && *item.Policy == *ingressRoute.Policy
+			return matches
+		}); err != nil {
+			return fmt.Errorf("failed adding static route: %v", err)
+		}
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) ensureRemoteZonePodAddressesToNodeRoute(pod *kapi.Pod) error {
+	// Cleanup from VM running at remote zone
```

Comment:
you mean cleanup from VM possibly running in local zone?

Reply:
right but the comment here does not say that. I would reword it to say what you just wrote: "DHCPOptions are only needed at the node is running the VM at that's the local zone node not the remote zone"